### ISSUE TITLE
block scoped variable capture errors is not shown if variable is closed in method of local class

### ISSUE THREAD

**Issue by vladima** [Microsoft/TypeScript#5143]
```ts
var data: any = [];
for (let x = 0; x < 10; ++x) {
    class C {
        m() {
            return x + 1;
        }
    }
    data.push(C);
}

for (let c of data) {
    console.log(new c().m());
}
```
This code is compiled with no errors and prints 11 10 times in runtime

**Comment by mhegazy** [comment]
is not this and #5144 mutually exclusive? either we make it work correctly or show the error? or am i missing something?

**Comment by vladima** [comment]
these issues are different (but both of them should work in sense that we should either report an error or do the right thing):
- in #5143 we don't account that local class can act as closure and thus don't report an error. Note that data here stores `C` and not `() => C` 
- in #5144 we miss the fact that class itself should behave as block-scoped variable that is captured in closure

**Comment by mhegazy** [comment]
> in #5143 we don't account that local class can create a closure (capture x) and thus don't report an error. Note that data here stores C and not () => C

Compiling the above code, i see the error as expected: 
> C:/test/file1.ts(2,1): error TS4091: Loop contains block-scoped variable 'x' referenced by a function in the loop. This is only supported in ECMAScript 6 or higher.

**Comment by vladima** [comment]
ok, something seems to be messed up with my enlistment. After trying on clean environment I can see that we do report an error.
